### PR TITLE
libbpf-cargo: gen: Add datasec mmap support

### PR DESCRIPTION
**Pull request by danobi**
This PR adds support for mmap'd datasec fields. Userspace can now
share data with progs via simple variable reads/writes. See the final
commit for the user facing interface.

See individual commits for details.

### CODE REVIEW

**Comment by danobi** [comment]
Stacked on top of #39

File: examples/runqslower/src/main.rs [hunk 1]
```diff
@@ -75,16 +75,13 @@ fn main() -> Result<()> {
...
-    let mut skel = skel_builder.open()?.load()?;
+    let mut open_skel = skel_builder.open()?;
 
     // Write latency value into map
-    skel.maps().min_us().update(
-        &0u32.to_le_bytes(),
-        &opts.latency.to_le_bytes(),
-        MapFlags::empty(),
-    )?;
+    open_skel.rodata().min_us = opts.latency;
```
Comment (anakryiko) [inline examples/runqslower/src/main.rs:81]: this looks pretty nice! really cool!

**Review by anakryiko — COMMENTED** [review]
How is this PR related to https://github.com/libbpf/libbpf-rs/pull/39 ? Is it a super-set?

It also seems like you haven't updated the skeleton, it still looks like the one from PR39, no?

**Comment by danobi** [comment]
> How is this PR related to #39 ? Is it a super-set?

Yes, this PR builds on top of the work in #39. I made this a separate PR b/c I thought it was complicated enough to merit it's own PR. I left it in draft mode to imply not to look at this yet.

**Comment by anakryiko** [comment]
oh, didn't know there is a draft state, I'll try to pay attention :)

**Comment by danobi** [comment]
Closing in favor of #43 .

**Pull request closed without merging**